### PR TITLE
[Messenger] fixed queue_name option on amazon sqs connection

### LINKED ISSUE #37293: [Messenger] queue_name option not used in amazon sqs connection
**Symfony version(s) affected**: 5.1

**Description**  
queue_name option not used in amazon sqs connection when setup transports the bridge create a queue with an empty name

**How to reproduce**
.env.dist
 
```MESSENGER_TRANSPORT_DSN=sqs://localhost:9324?sslmode=disable```

config/packages/messenger.yaml
```
framework:
    messenger:
        failure_transport: failed
        transports:
            failed:
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
                options:
                    queue_name: 'failed'
```

dump `$this->configuration` on Symfony\Component\Messenger\Bridge\AmazonSqs\Transport\Connection:231
```
array:11 [
    "buffer_size" => 9
    "wait_time" => 20
    "poll_timeout" => 0.1
    "visibility_timeout" => null
    "auto_setup" => false
    "access_key" => null
    "secret_key" => null
    "endpoint" => "https://sqs.eu-west-1.amazonaws.com"
    "region" => "eu-west-1"
    "queue_name" => ""
    "account" => null
]
```

**Possible Solution** 
the reason for the bug is not the queue name in the MESSENGER_DSN and the queue_name option is not used
 
*** suggestions ***
throw InvalidArgument when $parsedPath return empty queue_name and override queue_name if it's defined in options on  
Symfony\Component\Messenger\Bridge\AmazonSqs\Transport\Connection::fromDsn

**Additional context**  
I used [roribio16/alpine-sqs](https://hub.docker.com/r/roribio16/alpine-sqs/) to simulate sqs locally

### PR DESCRIPTION
**Pull request by ck-developer**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37293 
| License | MIT

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php [hunk 1]
```diff
@@ -115,6 +115,24 @@ public function testFromDsnWithQueryOptions()
...
+            new Connection(['queue_name' => 'queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
+            Connection::fromDsn('sqs://default', ['queue_name' => 'queue'], $httpClient)
+        );
+    }
+
+    public function testFromDsnWithOverridingQueueNameByOption()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['queue_name' => 'overrided_queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
```
Comment (jderusse) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php:131]: I agree with previous test but this one is wrong. For consistency with all other Symfony components: DSN must take precedence over options.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php [hunk 2]
```diff
@@ -119,7 +120,7 @@ public static function fromDsn(string $dsn, array $options = [], HttpClientInter
         }
 
         $parsedPath = explode('/', ltrim($parsedUrl['path'] ?? '/', '/'));
-        if (\count($parsedPath) > 0) {
```
Comment (jderusse) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:122]: previous code were buggy. explode always return an not empty array.
we should assert here that queue name is != ampty string

**Review by jderusse — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php [hunk 3]
```diff
@@ -124,6 +124,15 @@ public function testFromDsnWithQueryOptions()
...
     }
 
+    public function testFromDsnWithQueueNameOption()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['queue_name' => 'queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
+            Connection::fromDsn('sqs://default', ['queue_name' => 'queue'], $httpClient)
+        );
+    }
```
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php:134]:
```suggestion
        $this->assertEquals(
            new Connection(['queue_name' => 'my_queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
            Connection::fromDsn('sqs://default/my_queue', ['queue_name' => 'queue_ignored'], $httpClient)
        );
    }
```

**Review by Nyholm — COMMENTED** [review]
Thank you. 

What about adding a test for when you provide both a parameter and add a DSN with a queue name in it?

What should be the preferred result?

**Review by Nyholm — APPROVED** [review]
Thank you

**Comment by chalasr** [comment]
Good catch, thanks @ck-developer.

**Pull request merged**